src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHelper.Digest.cs

```diff
@@ -95,11 +95,7 @@ internal partial class AuthenticationHelper
                 return null;
             }
 
-            string opaque;
-            if (!digestResponse.Parameters.TryGetValue(Opaque, out opaque))
-            {
-                return null;
-            }
+            string opaque = digestResponse.Parameters.ContainsKey(Opaque) ? digestResponse.Parameters[Opaque] : null;
 
             string realm = digestResponse.Parameters.ContainsKey(Realm) ? digestResponse.Parameters[Realm] : string.Empty;
```

Comment:
From RFC, realm is required in digest-challenge, can you fix this to return null? Or you can file a bug, and i'll take care of it..

Reply:
done.